lib/nfc/helpers/mf_classic_dict.c

```diff
@@ -69,9 +69,15 @@ MfClassicDict* mf_classic_dict_alloc(MfClassicDictType dict_type) {
         string_t next_line;
         string_init(next_line);
         while(true) {
-            if(!stream_read_line(dict->stream, next_line)) break;
+            if(!stream_read_line(dict->stream, next_line)) {
+                FURI_LOG_T(TAG, "No keys left in dict");
+                break;
+            }
+            string_strim(next_line);
+            FURI_LOG_T(
+                TAG, "Read line: %s, len: %d", string_get_cstr(next_line), string_size(next_line));
             if(string_get_char(next_line, 0) == '#') continue;
-            if(string_size(next_line) != NFC_MF_CLASSIC_KEY_LEN) continue;
+            if(string_size(next_line) != NFC_MF_CLASSIC_KEY_LEN - 1) continue;
```

Comment:
I suggest changing NFC_MF_CLASSIC_KEY_LEN value since you add string_string() call